### ISSUE TITLE
Set Response Headers on API calls

### ISSUE THREAD

**Issue by ruudhaya** [GetStream/stream-go2#59]
While using Curl to getActivities, Response header contains values like 'rate_limit' of the particular API call, and 'remaining_rate_limit' after the current API call execution.
Same values are expected out of the GoLang Client Library.

**Comment by ferhatelmas** [comment]
Makes sense, will expose it as soon as possible.

**Comment by ferhatelmas** [comment]
I was thinking exposing headers in the `BaseResponse` but it will require changing multiple places since some calls don't return a response but only error. Then, headers would be a part of the error which isn't ideal since there are two places to check for it. Another option is having each call to return a response even if it's a basic. Actually, headers aren't a part of a response at all. It looks a hacky solution and it can be misleading to users which I don't like much. 

That's why cleaner solution, to me,  would be setting headers on the client and returning on demand such as `client.{Last}ResponseHeader()` which would return last headers. However, this will put state to the client which will prevent concurrent usage of the client. 

We can workaround this solution with a user provided channel such that client can push headers to this channel but then some bookkeeping is needed to map headers to requests which might complicate it further in concurrent case (which we're trying to solve). 

To sum up, the best solution is putting headers to the client and overwriting with the updated one, returning on demand, and also documenting client can be used concurrently unless headers are needed (we will have a protection for header). Otherwise, user can create multiple clients as needed and use them concurrently.

How does it sound to you ?

/cc @tbarbugli @ruudhaya

**Comment by ruudhaya** [comment]
I suggest having a proper solution with `BaseResponse`, since the hacky solutions will have too much overheads in handling concurrent usage
